mlir/utils/jenkins/Jenkinsfile

```diff
@@ -559,5 +560,51 @@ pipeline {
                 }
              }
         }
+        stage("Store a Release Build") {
+            when {
+                beforeAgent true;
+                equals expected: false, actual: params.weekly;
+                equals expected: false, actual: params.nightly;
+                not {equals expected: '', actual: params.releaseBranch}
+            }
+            agent {
+                docker {
+                    image dockerImage()
+                    args dockerArgs()
+                    label "${params.canXdlops ? 'gfx908' : 'rocm' }"
+                    alwaysPull true
+                }
+            }
+            stages{
+                stage ('Build and Archive a Release Branch') {
+                    steps {
+                        showEnv()
+                        sh "git checkout ${params.releaseBranch}"
+                        buildProject('', """
+                          -DBUILD_FAT_LIBMLIRMIOPEN=ON
+                          -DCMAKE_BUILD_TYPE=Release
+                          -DCMAKE_EXPORT_COMPILE_COMMANDS=1
+                        """)
+                        cmake arguments: "--install . --component libMLIRMIOpen --prefix ${WORKSPACE}/build/include",\
+                                installation: 'InSearchPath', workingDir: 'build'
+                        sh """#!/bin/bash
+                            temp=$params.releaseBranch
+                            release=\${temp////-}
+                            tar cvzf mlir-\$release.tar.gz ./build/bin ./build/lib ./build/include\
```

Comment:
To clarify - we're building a binary release? Wouldn't it make more sense to use something like `CPack` here so that, on top of getting a tarball, we could also do something like make a `.deb` automatically if at some point we wanted to?

Reply:
I'm not sold on the directory structure of this tarball - I'd want it to be `miir-x.y.z/{bin,include,lib,external}` and with a README and the like.

I'd also want to confirm that we don't end up with any concerning absolute paths in the resulting binaries.

Reply:
Good point. Let me try it. It's not clear yet how we are going to use the stored package, but we'll figure it out later.